test/e2e/specs/editor/various/block-renaming.spec.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Block Renaming', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.describe( 'Dialog renaming', () => {
+		test( 'allows renaming of blocks that support the feature via dialog-based UI', async ( {
+			editor,
+			page,
+			pageUtils,
+		} ) => {
+			// Create a two blocks on the page.
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: 'First Paragraph' },
+			} );
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: 'Second Paragraph' },
+			} );
+
+			// Multiselect via keyboard.
+			await pageUtils.pressKeys( 'primary+a' );
+			await pageUtils.pressKeys( 'primary+a' );
+
+			// Convert to a Group block which supports renaming.
+			await editor.clickBlockOptionsMenuItem( 'Group' );
+
+			await editor.clickBlockOptionsMenuItem( 'Rename' );
+
+			const renameMenuItem = page.getByRole( 'menuitem', {
+				name: 'Rename',
+				includeHidden: true, // the option is hidden behind modal but assertion is still valid.
+			} );
+
+			await expect( renameMenuItem ).toHaveAttribute(
+				'aria-expanded',
+				'true'
+			);
+
+			const renameModal = page.getByRole( 'dialog', {
+				name: 'Rename block',
+			} );
+
+			// Check focus is transferred into modal.
+			await expect( renameModal ).toBeFocused();
+
+			// Check the Modal is perceivable.
+			await expect( renameModal ).toBeVisible();
+
+			const saveButton = renameModal.getByRole( 'button', {
+				name: 'Save',
+				type: 'submit',
+			} );
+
+			await expect( saveButton ).toBeDisabled();
+
+			const nameInput = renameModal.getByLabel( 'Block name' );
+
+			await expect( nameInput ).toHaveAttribute( 'placeholder', 'Group' );
+
+			await nameInput.fill( 'My new name' );
+
+			await expect( saveButton ).toBeEnabled();
+
+			await saveButton.click();
+
+			await expect( renameModal ).toBeHidden();
+
+			// Check that focus is transferred back to original "Rename" menu item.
+			await expect( renameMenuItem ).toBeFocused();
+
+			await expect( renameMenuItem ).toHaveAttribute(
+				'aria-expanded',
+				'false'
+			);
+
+			await expect.poll( editor.getBlocks ).toMatchObject( [
+				{
+					name: 'core/group',
+					attributes: {
+						metadata: {
+							name: 'My new name',
+						},
+					},
+				},
+			] );
+		} );
+
+		test( 'allows custom name to be removed and reset to original block name', async ( {
```

Comment:
Non-blocking nit: e2e tests can be fairly slow to run. Would it be worth combining the logic in each of these tests into the one test, or is it preferable to keep them separate?

Reply:
Yes I agree. I've merged this into a single test 👍